pkg/internal/client/txn.go

```diff
@@ -686,7 +686,7 @@ func (txn *Txn) exec(ctx context.Context, fn func(context.Context, *Txn) error)
 	// Run fn in a retry loop until we encounter a success or
 	// error condition this loop isn't capable of handling.
 	for {
-		if err := ctx.Err(); err != nil {
+		if err := errors.WithStack(ctx.Err()); err != nil {
```

Comment:
would constructions like this more idiomatically be done with the errors.WithStack only on the next line? unclear.